.github/workflows/ci.yml

```diff
@@ -2,51 +2,63 @@ name: ci
 
 on: [push, pull_request]
 
+env:
+  FORCE_COLOR: 1
+
 jobs:
 
   pre-commit:
 
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
 
     - name: Install Python
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
       with:
         python-version: '3.10'
-        cache: pip
-        cache-dependency-path: pyproject.toml
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
+      with:
+        python-version: ${{ matrix.python-version }}
+        activate-environment: true
 
     - name: Install Python package and dependencies
-      run: pip install -e .[dev]
+      run: uv pip install -e .[dev]
 
     - name: Run pre-commit
       run: pre-commit run --all-files || ( git status --short ; git diff ; exit 1 )
 
   tests:
 
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     timeout-minutes: 10
 
     strategy:
+      fail-fast: false
       matrix:
         python-version: ['3.9', '3.10', '3.11', '3.12']
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
 
     - name: Install Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
```

Comment:
Why are you switching to uv in the CI but not the CD? I think it is good to keep the same approach for both, to minimize failures of CD even though CI is passing

Reply:
I added uv there as well, I didn't realize you're running tests in cd.yml.

Reply:
Really need to look sometime whether it is possible to use templates and get rid of the duplication. Is annoying to have to maintain both CI and CD workflows that are so similar